velox/exec/ContainerRowSerde.cpp

```diff
@@ -393,6 +393,44 @@ int compare<TypeKind::VARBINARY>(
   return flags.ascending ? result : result * -1;
 }
 
+template <TypeKind Kind>
+int compareFloatingPoint(
+    ByteStream& left,
+    const BaseVector& right,
+    vector_size_t index,
+    CompareFlags flags) {
+  VELOX_DCHECK(!flags.stopAtNull, "not supported compare flag");
+  using T = typename TypeTraits<Kind>::NativeType;
+  auto rightValue = right.asUnchecked<SimpleVector<T>>()->valueAt(index);
+  auto leftValue = left.read<T>();
+
+  int result{0};
+  if (std::isnan(leftValue) && std::isnan(rightValue)) {
+    result = 0;
+  } else {
+    result = leftValue < rightValue ? -1 : leftValue == rightValue ? 0 : 1;
+  }
+  return flags.ascending ? result : result * -1;
+}
+
+template <>
+int compare<TypeKind::DOUBLE>(
```

Comment:
There is another version of compare that takes 2 ByteStreams. Should it be modified as well?